src/plugins/intel_cpu/src/graph.cpp

```diff
@@ -1455,17 +1446,20 @@ void Graph::ParalleMtNuma(size_t num_nodes,
 }
 
 void Graph::Infer(SyncInferRequest* request) {
-    DEBUG_LOG("Starting inference of the graph: ", GetName(), ". Status: ", static_cast<int>(status));
-    if (!IsReady()) {
-        OPENVINO_THROW("Wrong state of the ov::intel_cpu::Graph. Topology is not ready.");
-    }
-
-    if (Status::ReadyDynamic == status) {
-        InferDynamic(request);
-    } else if (Status::ReadyStatic == status) {
+    DEBUG_LOG("Infer graph: ", GetName(), ". Status: ", static_cast<int>(status));
+
+    switch (status) {
+    case Status::ReadyDynamic:
+        InferDynamic(request, UpdateNodes(m_executableGraphNodes));
+        break;
+    case Status::ReadyDynamicSeq:
+        InferDynamic(request, UpdateNodesSeq(m_executableGraphNodes));
+        break;
+    case Status::ReadyStatic:
         InferStatic(request);
-    } else {
-        OPENVINO_THROW("Unknown ov::intel_cpu::Graph state: " , static_cast<size_t>(status));
+        break;
+    default:
+        OPENVINO_ASSERT(IsReady(), "Wrong state of the ov::intel_cpu::Graph. Topology is not ready: ", static_cast<int>(status));
```

Comment:
We can even drop this branch simply assigning a corresponding function pointer at the compilation stage, it will allow us keep the status enumeration simpler.

Reply:
Yes, this is the option I have also considered
As well as using an interface with multiple implementations.
But I think switch case is even more clean, explicit, lightweight and straightforward here and do not require strict alignment of infer strategies semantics.
If the list of the infer strategies grows big, we can try to replace it with a function pointer.